server/controllers/PoliticalParty.js

```diff
@@ -1,103 +1,129 @@
-    };
+  static async createParty(req, res) {
+    const createQuery = `INSERT INTO 
+      politicalparties(id, name, alias, hqaddress, logourl, datecreated)
+      VALUES($1, $2, $3, $4, $5, $6) returning *`;
 
-    politicalParties.push(party);
+    const values = [
+      uuid(),
+      req.body.name,
+      req.body.alias,
+      req.body.hqAddress,
+      req.body.logoUrl,
+      moment(new Date()),
+    ];
 
-    res.status(201).json({
-      status: 201,
-      message: 'Party Successfully Created',
-      data: [party],
-    });
+    try {
+      const { rows } = await dba.query(createQuery, values);
+      res.status(201).json({
+        status: 201,
+        message: 'Party Successfully Created',
+        data: [rows[0]],
+      });
+    } catch (error) {
+      return res.status(400).json({
+        status: 400,
+        message: error,
+      });
+    }
   }
 
-  static viewPartyById(req, res) {
+  static async viewPartyById(req, res) {
     const { partyid } = req.params;
-    const party = politicalParties.find(p => p.id === parseInt(partyid));
-    if (!party) {
-      return res.status(404).send({
+    const text = 'SELECT * FROM politicalparties WHERE id = $1';
+
+    try {
+      const { rows } = await dba.query(text, [partyid]);
+      return res.status(200).send({
+        status: 200,
+        data: [rows[0]],
+      });
+    } catch (error) {
+      return res.status(404).json({
         status: 404,
         error: 'The record with the given id was not found',
       });
     }
-    return res.status(200).json({
-      status: 200,
-      data: [party],
-    });
   }
 
-  static getAllParties(req, res) {
-    res.json({
-      status: 200,
-      data: politicalParties,
-    });
-  }
-
-  static deletePartyById(req, res) {
-    const { partyid } = req.params;
-    const findParty = politicalParties.find(x => x.id === parseInt(partyid));
-    if (!findParty) {
-      return res.status(404).json({
-        status: 404,
-        message: 'Party not found',
+  static async getAllParties(req, res) {
+    try {
+      const result = await dba.query('SELECT * FROM politicalparties');
+      return res.status(200).send({
+        status: 200,
+        data: result.rows,
+      });
+    } catch (error) {
+      return res.status(400).json({
+        status: 400,
+        error,
       });
     }
+  }
 
-    const index = politicalParties.indexOf(findParty);
-    const delparty = politicalParties.splice(index, 1);
-    if (!delparty) {
-      return res.status(400).json({
+
+  static async deletePartyById(req, res) {
+    const deleteQuery = 'DELETE FROM politicalparties WHERE id=$1 returning *';
+    try {
+      const { rows } = await dba.query(deleteQuery, [req.params.partyid]);
+      if (!rows[0]) {
+        return res.status(404).send({
+          status: 400,
+          message: 'Party not found',
+        });
+      }
+      return res.status(202).json({
+        status: 202,
```

Comment:
Kindly confirm if this is the right status code

Reply:
But your response message says delete was successful, which means the async request was carried out. `A 202 response is typically used for actions that take a long while to process. It indicates that the request has been accepted for processing, but the processing has not been completed.`